applications/luci-app-acme/htdocs/luci-static/resources/view/acme/acme.js

```diff
@@ -527,6 +603,98 @@ return view.extend({
 	}
 })
 
+function _isFqdn(domain) {
+	// Is not an IP i.e. starts from alphanumeric and has least one dot
+	return /[a-z0-9-]\..*$/.test(domain) && !/[0-9-]\..*$/.test(domain);
```

Comment:
Please provide a few examples of what should and should not match here.

Reply:
Not even 0a.net will match. Test it here: https://regexr.com/

your regex currently allow
`<single character of lowercase or 0-9>.anything including spaces` and do not allow an initial wildcard.

Suggestions: 
- split on `.` and examine each array element. 
- Test for IPv4 only - bail early in that case. 
- Test for IPv6 only - bail early in that case. 

This likely will not handle punycode correctly. So your best bet is to handle cases which are not accepted and reject based on those.

Reply:
Try:
_isFqdn('flying pig 0a.net')

Reply:
Try also:
_isFqdn('0a.big no no')